skimage/transform/registration.py

```diff
@@ -0,0 +1,201 @@
+        determines the size of the image pyramid used. Choosing a smaller value
+        here can cause registration errors, but a larger value could speed up
+        registration when the alignment is easy.
+    multichannel : bool, optional
+        Whether the last axis of the image is to be interpreted as multiple
+        channels or another spatial dimension. By default, this is False.
+    inverse : bool, optional
+        Whether to return the inverse transform, which converts coordinates
+        in the reference space to coordinates in the target space. For
+        technical reasons, this is the transform expected by
+        ``scipy.ndimage.affine_transform`` to map the target image to the
+        reference space.
+    level_callback : callable, optional
+        If given, this function is called once per pyramid level with a tuple
+        containing the current downsampled image, transformation matrix, and
+        cost as the argument. This is useful for debugging or for plotting
+        intermediate results during the iterative process.
+    method : string
+        Method of minimization.  See ``scipy.optimize.minimize`` for available
+        options.
+    **kwargs : keyword arguments
+        Keyword arguments passed through to ``scipy.optimize.minimize``
+
+    Returns
+    -------
+    matrix : array
+        A transformation matrix used to obtain a new image.
+        ndi.affine_transform(target, matrix) will align your target image.
+
+    Example
+    -------
+    >>> from skimage.data import astronaut
+    >>> reference_image = astronaut()[..., 1]
+    >>> r = -0.12  # radians
+    >>> c, s = np.cos(r), np.sin(r)
+    >>> matrix_transform = np.array([[c, -s, 0], [s, c, 50], [0, 0, 1]])
+    >>> moving_image = ndi.affine_transform(reference_image, matrix_transform)
+    >>> matrix = register_affine(reference_image, moving_image)
+    >>> registered_moving = ndi.affine_transform(moving_image, matrix)
+    """
+
+    # ignore the channels if present
+    ndim = reference_image.ndim if not multichannel else reference_image.ndim - 1
+    if ndim == 0:
+        raise ValueError(
+            'Input images must have at least 1 spatial dimension.'
+        )
+
+    min_dim = min(reference_image.shape[:ndim])
+    nlevels = int(np.floor(np.log2(min_dim) - np.log2(pyramid_minimum_size)))
+
+    pyramid_ref = pyramid_gaussian(reference_image, downscale=pyramid_scale,
+                                   max_layer=nlevels,
+                                   multichannel=multichannel)
+    pyramid_mvg = pyramid_gaussian(moving_image, downscale=pyramid_scale,
+                                   max_layer=nlevels,
+                                   multichannel=multichannel)
+    image_pairs = reversed(list(zip(pyramid_ref, pyramid_mvg)))
+
+    if initial_vector is None:
+        initial_vector = np.eye(ndim, ndim + 1).ravel()
+    parameter_vector = initial_vector
+
+    if vector_to_matrix is None:
```

Comment:
do we really need these parameters? `vector_to_matrix` and `translation_indices`
This function has a lot of parameters, and it's easier to add some later on that to remove existing ones through a deprecation cycle...

Reply:
`vector_to_matrix` is essential — see the gallery example. (But possibly poorly-named.) And `translation_indices` then follows... =\